src/shared.js

```diff
@@ -1454,25 +1459,71 @@ export const updateCollectionSettingData = async (biospecimenData, tubes, partic
             }
         });
     }
+    else if (settings[visit][conceptIds.baseline.urineCollectedTime] !== '' ||  settings[visit][conceptIds.clinicalDashboard.urineCollectedTime] !== ''){
+        urineTubes.forEach(tube => {
+            const tubeIsNotCollected = biospecimenData[tube.concept][conceptIds.collection.tube.isCollected] === conceptIds.no;
+            const participantUrineCollected = participantData[conceptIds.baseline.urineCollected] === conceptIds.yes;
+            const collectionSetting = biospecimenData[conceptIds.collection.collectionSetting];
+            const isResearch = collectionSetting === conceptIds.research;
+            const isClinical = collectionSetting === conceptIds.clinical;
+
+            if(tubeIsNotCollected && participantUrineCollected) {
+                delete settings[visit][conceptIds.urineCollectionSetting];
+                if(isResearch) {
+                    delete settings[visit][conceptIds.baseline.urineCollectedTime];
+                }
+                else if(isClinical) {
+                    settings[visit][conceptIds.clinicalDashboard.urineCollected] = conceptIds.no;
+                    delete settings[visit][conceptIds.clinicalDashboard.urineCollectedTime];
+                    settings[visit][conceptIds.anySpecimenCollected] = conceptIds.no;
```

Comment:
is this still valid if a blood tube has been collected?

Reply:
Let me rephrase... If a blood tube is collected and a urine tube is accidentally collected, when we go back and uncheck the urine tube wouldn't we still want `anySpecimenCollected` to be `yes` ?